src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colSup = new ColorSupplier();
+    private Figure figure;
+    private Shape shape;
+
+    public Shape getRandomFigure() {
+
+        int i = random.nextInt(Figure.values().length);
+        figure = Figure.values()[i];
+
+        switch (figure) {
+            case CIRCLE:
+                shape = new Circle(colSup.getRandomColor(), random.nextInt(100));
+                break;

```

Comment:
```suggestion
                return getRandomCircle();
```
lets add private methods in this class to get each figure - this would make your code easier to read